src/shared/hostname-setup.c

```diff
@@ -229,7 +231,11 @@ int hostname_setup(bool really) {
         if (really)
                 hostname_update_source_hint(hn, source);
 
-        return r;
+finish:
+        if (!in_initrd())
```

Comment:
If `!really`, I don't think we should send it out?

Reply:
i had that in place originally. But the only case `really` is set to false is in a testcase, and given that $NOTIFY_SOCKET is not going to be set there I figured there's no reason to special case this. And even if people call the tests with $NOTIFY_SOCKET set, I think it's entirely fine to generate the notification, maybe even people want to test for that part.

or in other words: given that this notification message isn#t destructive on its own, but just additive I see no reason to suppress that. And IRL it's going to have no effect anyway.